gbpservice/nfp/configurator/drivers/firewall/paloalto/paloAlto_fw_driver.py

```diff
@@ -0,0 +1,926 @@
+                                        == const.STATUS_ERROR:
+                return const.STATUS_ERROR
+
+        # commit
+        try:
+            resp = self.commit(hostname=vm_mgmt_ip)
+        except pan.xapi.PanXapiError as err:
+            self._print_exception('PanXapiError', err, const.COMMIT)
+            raise pan.xapi.PanXapiError(err)
+        except pan.config.PanConfigError as err:
+            self._print_exception('PanConfigError', err, const.COMMIT)
+            raise pan.config.PanConfigError(err)
+        except Exception as err:
+            self._print_exception('UnexpectedError', err, const.COMMIT, resp)
+            raise Exception(err)
+
+        if self.analyze_response("COMMITED the configuration to Service VM",
+                 resp, const.COMMIT) == const.STATUS_ERROR:
+            return const.STATUS_ERROR
+
+        return const.STATUS_ACTIVE
+
+    def update_firewall(self, context, firewall, host):
+        """ Implements firewall updation
+
+        Issues REST call to service VM for firewall updation
+
+        :param context: Neutron context
+        :param firewall: Firewall resource object from neutron fwaas plugin
+        :param host: Name of the host machine
+
+        Returns: UPDATED/ERROR status
+
+        """
+
+        vm_mgmt_ip = self._get_firewall_attribute(firewall)
+        msg = ("Initiating UPDATE request.")
+        LOG.info(msg)
+    
+        """
+        # update security rule
+        for curr_rule in firewall['firewall_rule_list']:
+            rule_name = curr_rule['name']
+            element = self.create_security_rule_element(
+                        rule_name=rule_name, action=curr_rule['action'],
+                        source_member=curr_rule['source_ip_address'],
+                        destination_member=curr_rule['destination_ip_address'])
+            try:
+                resp = self.edit_security_rule(vm_mgmt_ip, rule_name, element)
+            except pan.xapi.PanXapiError as err:
+                self._print_exception('PanXapiError', err,
+                                      const.UPDATE_SECURITY_RULE)
+                raise pan.xapi.PanXapiError(err)
+            except pan.config.PanConfigError as err:
+                self._print_exception('PanConfigError', err,
+                                      const.UPDATE_SECURITY_RULE)
+                raise pan.config.PanConfigError(err)
+            except Exception as err:
+                self._print_exception('UnexpectedError', err,
+                                      const.UPDATE_SECURITY_RULE, resp)
+                raise Exception(err)
+
+            if self.analyze_response("UPDATED the configuration to Service VM",
+                                     resp, const.UPDATE_SECURITY_RULE)\
+                                        == const.STATUS_ERROR:
+                return const.STATUS_ERROR
+        """
+
+        # commite security rule
```

Comment:
Is update a No OP, as we are not doing anything for update ?
If yes, then why do we need to do commit ?